src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/HeadersConfigurerLambdaDslTest.java

```diff
@@ -38,79 +38,157 @@ void simpleContentSecurityPolicy() {
+              import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+              import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+
+              @Configuration
+              public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+                  @Override
+                  protected void configure(HttpSecurity http) throws Exception {
+                      http
+                              .headers()
+                              .xssProtection(protection -> protection.and());
+                  }
+              }
               """
           )
         );
     }
 
     @Test
-    void complexContentSecurityPolicy() {
+    void xssProtectionDisable() {
         //language=java
         rewriteRun(
           java(
             """
-import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
-import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+              import org.springframework.context.annotation.Configuration;
+              import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+              import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 
-@EnableWebSecurity
-public class ConventionalSecurityConfig extends WebSecurityConfigurerAdapter {
+              @Configuration
+              public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+                  @Override
+                  protected void configure(HttpSecurity http) throws Exception {
+                      http
+                        .headers()
+                        .xssProtection().xssProtectionEnabled(false);
+                  }
+              }
+              """,
+            """
+              import org.springframework.context.annotation.Configuration;
+              import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+              import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
-        http
-                .headers(headers -> headers
-                        .contentSecurityPolicy("foobar").reportOnly());
+              @Configuration
+              public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+                  @Override
+                  protected void configure(HttpSecurity http) throws Exception {
+                      http
+                              .headers()
+                              .xssProtection(protection -> protection.disable());
+                  }
+              }
+              """
+          )
+        );
     }
-}
+
```

Comment:
There's potentially an edge case related to the`XssProtectionSpec.headerValue` added in Spring Security 5.8. Not a lot of folks are likely to use that, but figured call it out here as something that we either want to be explicit about not handling yet, or handle and test as well.

Reply:
It's supported by the origin recipe. I added a new test for it